pkg/plugins/document/firestore/firestore.go

```diff
@@ -83,13 +83,53 @@ func (s *FirestoreDocService) Delete(key *sdk.Key) error {
 
 	doc := s.getDocRef(key)
 
+	// 	Delete any sub collection documents
+	iter := doc.Collections(s.context)
+	for {
+		subColRef, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("error deleting value: %v", err)
+		}
+
+		for {
+			iter := subColRef.Limit(100).Documents(s.context)
```

Comment:
Could we rename this inner iterator variable to avoid confusion with the collections iterator on line 87?

Reply:
Good point, thx

Reply:
Have done